modules/apps/translation/translation-web/test/js/translate/Translate.js

```diff
@@ -0,0 +1,310 @@
+		sourceLanguageId: 'en_US',
+		targetLanguageId: 'es_ES',
+	},
+	autoTranslateEnabled: true,
+	getAutoTranslateURL: 'http://translation-url/auto_translate',
+	infoFieldSetEntries: [
+		{
+			fields: [
+				{
+					editorConfiguration: null,
+					html: false,
+					id: 'infoField--title--',
+					label: 'Title',
+					multiline: false,
+					sourceContent: 'mock title',
+					sourceContentDir: 'ltr',
+					targetContent: 'mock title',
+					targetContentDir: 'ltr',
+					targetLanguageId: 'es_ES',
+				},
+				{
+					editorConfiguration: {},
+					html: true,
+					id: 'infoField--description--',
+					label: 'Description',
+					multiline: false,
+					sourceContent: '<p>mock summary</p>',
+					sourceContentDir: 'ltr',
+					targetContent: '<p>mock summary</p>',
+					targetContentDir: 'ltr',
+					targetLanguageId: 'es_ES',
+				},
+			],
+			legend: 'Basic Information',
+		},
+		{
+			fields: [
+				{
+					editorConfiguration: {},
+					html: true,
+					id: 'infoField--content--',
+					label: 'Content',
+					multiline: true,
+					sourceContent: '<p>mock conent</p>',
+					sourceContentDir: 'ltr',
+					targetContent: '<p>mock conent</p',
+					targetContentDir: 'ltr',
+					targetLanguageId: 'es_ES',
+				},
+			],
+			legend: 'Content (Basic Web Content)',
+		},
+	],
+	portletId: 'mock_TranslationPortlet',
+	portletNamespace: '_mock_TranslationPortlet_',
+	publishButtonDisabled: false,
+	publishButtonLabel: 'Publish',
+	redirectURL: 'http://redirect-url',
+	saveButtonDisabled: false,
+	saveButtonLabel: 'Save as Draft',
+	sourceLanguageId: 'en_US',
+	sourceLanguageIdTitle: 'en-US',
+	targetLanguageId: 'es_ES',
+	targetLanguageIdTitle: 'es-ES',
+	translateLanguagesSelectorData: {
+		currentUrl: 'http://current-url',
+		sourceAvailableLanguages: ['en_US', 'es_ES'],
+		sourceLanguageId: 'en_US',
+		targetAvailableLanguages: [
+			'ar_SA',
+			'ca_ES',
+			'zh_CN',
+			'nl_NL',
+			'fi_FI',
+			'fr_FR',
+			'de_DE',
+			'hu_HU',
+			'ja_JP',
+			'pt_BR',
+			'es_ES',
+			'sv_SE',
+		],
+		targetLanguageId: 'es_ES',
+	},
+	translationPermission: true,
+	updateTranslationPortletURL: 'http://update-url',
+	workflowActions: {
+		PUBLISH: '1',
+		SAVE_DRAFT: '2',
+	},
+};
+
+const renderComponent = (props) => render(<Translate {...props} />);
+
+describe('Translate', () => {
+	Liferay.Util.sub.mockImplementation((langKey, ...args) =>
+		[langKey, ...args].join('-')
+	);
+
+	afterEach(cleanup);
+
+	it('renders with auto-translate enabled', () => {
+		const {asFragment} = renderComponent(baseProps);
+
+		expect(asFragment()).toMatchSnapshot();
+	});
+
+	it('renders auto-translate field button disabled when the sourceContent is empty', () => {
+		const {getByText} = renderComponent({
+			...baseProps,
+			infoFieldSetEntries: [
+				{
+					...baseProps.infoFieldSetEntries[1],
+					fields: [
+						{
+							...baseProps.infoFieldSetEntries[1].fields[0],
+							sourceContent: '',
+						},
+					],
+				},
+			],
+		});
+
+		expect(
```

Comment:
It is not easier to just check the snapshot (where I guess there will be a disabled button) rather than having to look for a node by its (language key) text, and then for the closest button?

Reply:
Isn't easier for me, it's expensive to update snapshots and in my experience, we end updating the snapshots without care about the changes.

This way is more specific and in this test, we only are interested in this button.

The selector isn't the best option https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-the-wrong-query but it's good enough. The best option `getByRole('button', {name: /auto-translate-x-field-.../i})` isn't available in the current version of `@testing-library/react`

Reply:
>we end updating the snapshots without care about the changes.

IMHO that shouldn't be a valid reason, we should care about everything we change. If not, we should get rid of ALL snapshots.

Doing this we're making the tests dependant on HTML markup, so if markup changes you need to manually update the tests to fit with the new case (in this case 1, but it could be tons of it). Using snapshots you just need to check if that markup change jest is complaining about is something expected and automatically update it.

This is a non blocker, just dropping my 2 cents.

Reply:
Yep, you are correct, we should care about everything we change, and IMHO the best we can do to afford this avoids large snapshots for little specific test like this.